### PR TITLE
Carthage and CocoaPods support

### PR DESCRIPTION
**Pull request by guidomb**
This is first attempt to add support for CocoaPods. It is currently not working because CocoaPods requires a valid semver number (and a corresponding git tag) as the library version.

This is based on #305 and should be merged after that PR is merged.

@emilsjolander @dshahidehpour Please check if the author information in the `YogaKit.podspec` file is correct.

Once this PR is merged in order for people to actually be able to install this using CocoaPods, the pod must be published to CocoaPods' trunk. Which requires to have an account. For more information check https://guides.cocoapods.org/making/getting-setup-with-trunk.html.

Also keep in mind that every time you make a new release you need to update the `version` property inside `YogaKit.podspec` and publish a new version to trunk.

### CODE REVIEW

File: YogaKit/YogaKit.podspec [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Pod::Spec.new do |spec|
+  spec.name                 = 'YogaKit'
+  spec.version              = 'v2016.12.26.00'
+  spec.license              = { :type => 'BSD-3-Clause', :file => '../LICENSE' }
+  spec.homepage             = 'https://facebook.github.io/yoga/'
+  spec.authors              = {
+                              'Emil Sjölander'      => 'sjolander.emil@gmail.com',
+                              'Dustin Shahidehpour' => 'email@email.com'
```
Comment (dshahidehpour) [inline YogaKit/YogaKit.podspec]: dshahidehpour@gmail.com ;)

File: YogaKit/YogaKit.podspec [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+                            }
+  spec.summary              = 'Yoga is a cross-platform layout engine which implements Flexbox.'
```
Comment (dshahidehpour) [inline YogaKit/YogaKit.podspec]: YogaKit is a view layout library which is powered by Yoga, a layout engine which implements Flexbox.

File: YogaKit/YogaKitSample/AppDelegate.h [hunk 3]
```diff
@@ -12,5 +12,6 @@
 
 @property (strong, nonatomic) UIWindow *window;
 
+
```
Comment (dshahidehpour) [inline YogaKit/YogaKitSample/AppDelegate.h]: nit: delete

File: YogaKit/YogaKitSample/AppDelegate.m [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE-examples file in the root directory of this source tree.
+ */
+
+#import "AppDelegate.h"
```
Comment (dshahidehpour) [inline YogaKit/YogaKitSample/AppDelegate.m]: Do we have to regenerate this? Let's revert this.

File: YogaKit/YogaKit.podspec (hunk 2 above)
Comment (guidomb) [inline YogaKit/YogaKit.podspec]: I've copy pasted that description from the one that appears in Yoga's GitHub repository. Should I change it anyways?

**Comment by guidomb** [comment]
@dshahidehpour You made some comments in files that are originally part of @hartbit PR, #305 . For some reason I could not make  `hartbit/carthage-support` as the target branch for this PR (GitHub did not show @hartbit's fork as an option and I couldn't fork his repo because I've already had a fork of yoga). I think those suggestion should be addressed in #305. Sorry for the extra work in the review process

**Comment by hartbit** [comment]
@guidomb but as you improved on my PR, wouldn't it be better to rename this PR Cocoapods + Carthage support and fix everything here?

**Comment by dshahidehpour** [comment]
@guidomb Any update on this? The changes to add swift support have landed, so, you probably won't run into as many (if any) rebase problems.

**Comment by guidomb** [comment]
@dshahidehpour Sorry, I've been on vacations the last week. Cool that #322 has been merged!!! I'll try to get this working tomorrow. I'll let you know if I had any issues.

Have you defined a version number that should be used in the CocoaPods podspec file?

**Comment by emilsjolander** [comment]
@guidomb No defined a version yet. I'll do that tomorrow! Thanks for reminding me. Will most likely be version `1.0.0` but i'll have to discuss it with some others.

**Comment by emilsjolander** [comment]
@guidomb lets wait on #328 to merge this.

**Comment by guidomb** [comment]
OK, I was about to start working on it. I'll subscribe to #328 and get back to this once that one is merged. Thanks for letting me know!

**Comment by dshahidehpour** [comment]
@guidomb #328 is going to be closed momentarily, any chance we could start on this again? @emilsjolander can help with the versioning :)

**Comment by guidomb** [comment]
I'll try to work on this tomorrow. Will #328 be merged into `master`?

**Comment by dshahidehpour** [comment]
yeahp!

**Comment by guidomb** [comment]
I've made a new PR to address this -> https://github.com/facebook/yoga/pull/352

**Pull request closed without merging**